guidance/_grammar.py

```diff
@@ -966,10 +966,6 @@ def _re_with_temperature(grammar, temperature, visited_set):
 #     return ModelVariable(name)
 
 
-def active_role_end() -> ModelVariable:
```

Comment:
The current implementation of `active_role_end` will actually produce the closing tag twice since it is essentially a stateless grammar and can't pop `ContextBlock`s off of a model. Furthermore, generating active_role_end requires the model to produce special tags which will be hidden from future prompts... A better implementation of this may exist, but I'm not quite sure yet. Deleting it for now